session/bootstrap.go

```diff
@@ -1835,6 +1835,15 @@ func upgradeToVer89(s Session, ver int64) {
 // to the error log. The message is important since the behavior is weird
 // (changes to the config file will no longer take effect past this point).
 func importConfigOption(s Session, configName, svName, valStr string) {
+	if valStr == "" || valStr == "0" {
```

Comment:
There will be a little chance that '0' is explicitly specified in the configuration file, but IMHO it's fine to skip it here.

Reply:
That's right. It will need to be fixed if we import a possible 0-value config item in future.

i.e. For most cache-settings we usually have a flag for enabled, and another flag for capacity. But it's common (outside of tidb) that you just have one flag for capacity, and 0 means not enabled. If this case existed it might not be handled correctly.